datafusion/core/src/physical_optimizer/optimizer.rs

```diff
@@ -89,6 +90,8 @@ impl PhysicalOptimizer {
             // as that rule may inject other operations in between the different AggregateExecs.
             // Applying the rule early means only directly-connected AggregateExecs must be examined.
             Arc::new(LimitedDistinctAggregation::new()),
+            // Run once before PartialFinalAggregation is rewritten to ensure the rule is applied correctly
+            Arc::new(OptimizeAggregateOrder::new()),
```

Comment:
I think we can remove this rule from here. Using it only in below place should be enough.